prow/gerrit/client/client.go

```diff
@@ -246,11 +257,12 @@ func (c *Client) QueryChanges(lastState LastSyncState, rateLimit int) map[string
 	for _, h := range c.handlers {
 		lastStateForInstance := lastState[h.instance]
 		changes := h.queryAllChanges(lastStateForInstance, rateLimit)
-		if len(changes) > 0 {
-			result[h.instance] = []ChangeInfo{}
-			for _, change := range changes {
-				result[h.instance] = append(result[h.instance], change)
-			}
+		if len(changes) == 0 {
+			continue
+		}
+
+		for _, change := range changes {
+			result[h.instance] = append(result[h.instance], change)
```

Comment:
Is it safe to assume `h.instance` always exists in the `result` map? In the deleted code there is `result[h.instance] = []ChangeInfo{}`

Reply:
That is not necessary.

```golang
var foo []string = nil
foo = append(foo, "hello")
```

works just fine. This is in fact one of the reasons why append returns a value.

See https://play.golang.org/p/EVcqaQ0JNOI

Reply:
I see. Thanks for the explanation and example!